### PR TITLE
[MM-33679] Fix Accessibility for Custom Status and Modals

### PR DESCRIPTION
**Pull request by mylonsuren**
#### Summary

Adds voiceover accessibility support for custom status and modal headers.

#### Ticket Link

Fixes [MM-33679](https://mattermost.atlassian.net/browse/MM-33679)

#### Screenshots

| before | after |
|----|----|
| imageimage | Screen Shot 2022-02-10 at 6 55 36 PMScreen Shot 2022-02-10 at 6 58 24 PM|

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by mkraft — APPROVED** [review]

File: components/generic_modal.tsx [hunk 1]
```diff
@@ -130,7 +131,8 @@ export default class GenericModal extends React.PureComponent<Props, State> {
                 enforceFocus={this.props.enforceFocus}
                 restoreFocus={true}
                 role='dialog'
-                aria-labelledby='genericModalLabel'
+                aria-label={this.props.ariaLabel}
+                aria-labelledby={this.props.ariaLabel ? undefined : 'genericModalLabel'}
```
Comment (BenCookie95) [inline components/generic_modal.tsx:135]: Could do `this.props.ariaLabel || 'genericModalLabel'`, 0/5

**Review by BenCookie95 — APPROVED** [review]
LGTM! Nice work

File: components/generic_modal.tsx (hunk 1 above)
Comment (mylonsuren) [reply components/generic_modal.tsx:135]: I would've preferred do to something like that as well, but it guarantees that `aria-labelledby` is always defined which would take priority over `aria-label`. The issue stems from `aria-labelledby` not being able to read the value from the `FormattedMessage` component that is used for the header, so I just give priority to `aria-label` for components that use it to resolve the issue.

**Review by furqanmlk — APPROVED** [review]
Working as expected. Test case has been created [MM-T4654](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4654)

image

**Pull request merged**